lib/widget/paragraph.cpp

```diff
@@ -558,3 +578,24 @@ void Paragraph::highlightLost()
 	isMouseDown = false;
 	dirty = true;
 }
+
+nonstd::optional<std::vector<uint32_t>> Paragraph::getScrollSnapOffsets()
```

Comment:
Can be made `const`.

Reply:
Ah, I see. So, it's a virtual method and this can possibly involve some layout updating, meaning this is ok.